### DISCUSSION TITLE
百字说清 EVM++ 的本质

### DISCUSSION THREAD

**Discussion by ourai** [openbuildxyz/Web3-Frontend-Bootcamp#1095] [Materials]
在官网的《[EVM++基础概念](https://openbuild.xyz/learn/challenges/2036589711/1718782513)》中看了一点介绍后，给我的第一感觉就是造了个 WASM VM 去监听 EVM 事件/消息，在特定的几个事件/消息处理函数中执行开发者开发的扩展 hook 队列。

之后所看的课程其他部分文字内容，基本就是在验证我的猜想；即便有所出入，整体上也是大差不差。

看到「EVM++」这个字眼，直觉上就给人是「对 EVM 进行加强」的印象，实际上也确实如此：

- 其中一个「+」是倚赖并行计算与弹性区块空间提升规模上的可扩展性（scalability）；
- 另外一个「+」是通过 WASM VM 提高功能上的可扩展性（extensibility）。

即 `EVM++ = EVM + WASM VM + connector(EVM, WASM VM)`，是 [Artela](https://artela.network/) 这条 L1 公链的基础与卖点，而「Aspect」则是其支撑模块化扩展的核心要素。

初步了解下来，我觉得「Aspect」这个命名挺直白的——践行[面向切面编程（AOP）](https://en.wikipedia.org/wiki/Aspect-oriented_programming)，把切面作为「一级公民」，其开发就是编写一个个切面，实现各个 hook 的逻辑就完了。

因为写 Aspect 用的是类 TypeScript 的 [AssemblyScript](https://www.assemblyscript.org/)，开发体验会与写 [Vue Class Component](https://class-component.vuejs.org/) 很相似。

**Comment by dinhcuong261291** [comment]
GM++
